addons/sale_mrp/tests/test_sale_mrp_procurement.py

```diff
@@ -363,3 +375,47 @@ def test_sale_mrp_avoid_multiple_pickings(self):
         so.action_confirm()
         self.assertEqual(len(so.picking_ids), 1)
         self.assertEqual(so.picking_ids.picking_type_id, warehouse.out_type_id)
+
+    def test_routes_on_sale_order_line(self):
+        """ Ensure that when multiple routes are available on a product and sale order line,
+        the system considers routes on the Sale Order Line (SOL) and respects their priority over product routes.
+
+        For example, if MTO + Buy is selected on sale order line(while the product has a Manufacture route),
+        a Purchase Order should be created instead of triggering Manufacture.
+        """
+        if self.env['ir.module.module']._get('purchase').state != 'installed':
+            self.skipTest("purchase module is not installed.")
+
+        warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        mrp_route = warehouse.manufacture_pull_id.route_id
+        buy_route = warehouse.buy_pull_id.route_id
+        mto_route = warehouse.mto_pull_id.route_id
+        mto_route.active = True
+
+        # Make routes selectable for sales.
+        mrp_route.sale_selectable = buy_route.sale_selectable = True
+
+        partner_1, partner_2 = self.env['res.partner'].create([{'name': 'partner_1'}, {'name': 'partner_2'}])
+        component = self.env['product.product'].create([{
+            'name': 'component',
+            'is_storable': True,
+            'route_ids': [Command.set([mrp_route.id, buy_route.id, mto_route.id])],
+            'seller_ids': [Command.create({'partner_id': partner_1.id})],
+        }])
+
+        # Create a sale order without specified the routes on sale order line.
+        so1 = self._create_sale_order(component, 1, partner_2)
+        so1.action_confirm()
+        # MO is created for this product, as both mrp and buy route are selected on product with mto, but mrp route has higher priority so that mo is created.
+        mrp_order = self.env['mrp.production'].search([('reference_ids', 'in', so1.stock_reference_ids.ids)])
+        purchase_order = self.env['purchase.order'].search([('reference_ids', 'in', so1.stock_reference_ids.ids)])
+        self.assertEqual(len(mrp_order), 1)
+        self.assertEqual(len(purchase_order), 0)
+        # Create a sale order specified the routes on sale order line.
+        so2 = self._create_sale_order(component, 1, partner_2, [mto_route, buy_route])
```

Comment:
```suggestion
        so2 = self._create_sale_order(component, 1, partner_2, [buy_route])
```
Do you need to set `mto_route` here, as it's already defined on the product?